lib/linter/linter.js

```diff
@@ -808,6 +809,8 @@ function parse(text, languageOptions, filePath) {
         const visitorKeys = parseResult.visitorKeys || evk.KEYS;
         const scopeManager = parseResult.scopeManager || analyzeScope(ast, languageOptions, visitorKeys);
 
+        debug("Parsing successful:", filePath);
```

Comment:
That's a good point, I think it might be useful to have time taken for both steps logged separately. Could you hint how this could be solved generically that would handle built-in parser as well as `@typescript-eslint/parser`?

Reply:
> how this could be solved generically that would handle built-in parser as well as `@typescript-eslint/parser`

We can add one pair of debug logs around `const parseResult = ...`, and another pair around `const scopeManager = ...`.

When the built-in parser is used, that would log parse time and scope time separately.

When `@typescript-eslint/parser` is used, parse time and scope time would be both included in the first log (the second would be ~0 time) and it doesn't seem there's anything we can do to split them here. The only solution might be to ask `@typescript-eslint/parser` to add logs around [this line](https://github.com/typescript-eslint/typescript-eslint/blob/877cc48590c92cba7ff04c0cf421af50de9acab5/packages/parser/src/parser.ts#L173). Then, it would be possible to see `parse + scope` time (logged by ESLint) and `scope` time (logged by `@typescript-eslint/parser`) ,

Reply:
> The only solution might be to ask `@typescript-eslint/parser` to add logs around [this line](https://github.com/typescript-eslint/typescript-eslint/blob/877cc48590c92cba7ff04c0cf421af50de9acab5/packages/parser/src/parser.ts#L173). Then, it would be possible to see `parse + scope` time (logged by ESLint) and `scope` time (logged by `@typescript-eslint/parser`) ,

In this case, since both packages are using the same [`debug`](https://www.npmjs.com/package/debug) library, you could set environment variable `DEBUG=eslint:linter,typescript-eslint:parser:parser` and run eslint without `--debug` to see logs from both modules.

Reply:
Thanks @mdjermanovic, I applied your suggestions